doc/cabal-commands.rst

```diff
@@ -607,6 +607,22 @@ you will have access to its quickjump index.
 The generated landing page will contain one tree of all modules per local
 package.
 
+cabal check
+-----------
+
+``cabal check [FLAGS]`` checks the package for common mistakes (e.g.: if
+it is missing important fields like ``synopsis``, if it is using
+tricky GHC options, etc.).
+
+Run ``cabal check`` where your ``.cabal`` package file is.
+
+.. option:: -v, --verbose[=n]
+
+    Set verbosity level (0–3, default is 1).
+
+``cabal check`` mimics Hackage's requirements: if no error or warning
+is reported, Hackage will accept your package.
```

Comment:
Perhaps with "should" it would be more apparent the user should ignore cabal and file an error report if Hackage complains regardless.